sml/src/sml_transport.c

```diff
@@ -73,19 +55,22 @@ size_t sml_transport_read(int fd, unsigned char *buffer, size_t max_len) {
 	}
 
 	// found start sequence
-
 	while ((len+8) < max_len) {
+		bytes = read(fd, &(buf[len]), 4);
+		if (bytes < 4) {
+			return -1;
```

Comment:
It's not an error if read returns less than the number of bytes requested! (Even with blocking io. It blocks for some data not for all data.) With this code you risk that you loose data on fast machines. (see man 2 read )